sphinx/environment/__init__.py

```diff
@@ -351,17 +351,18 @@ def merge_info_from(self, docnames, other, app):
 
     def path2doc(self, filename):
         # type: (unicode) -> unicode
-        """Return the docname for the filename if the file is document.
+        """Return the docname for the filename if the file is document."""
+        if not path.isfile(filename):
+            return None
 
-        *filename* should be absolute or relative to the source directory.
```

Comment:
> Why removing this behavior?

Because we handle this elsewhere - look at https://github.com/sphinx-doc/sphinx/pull/4560/files#diff-bb8897b780588faa431faef0abc85e97R309 for example. The purpose of this function is to convert a path to a docname. Nothing more.



> What is returned for path2doc('/path/to/srcdir/index.rst')?
> I expect the return value is index. Does this works right?

Yes, this is what happens.

Reply:
Ah, sorry. I'd overlooked `path.relpath()`. Please forget my second comment.